src/swap/components/SwapProvider.tsx

```diff
@@ -52,16 +52,37 @@ export function SwapProvider({
   const { useAggregator } = experimental;
   // Core Hooks
   const accountConfig = useConfig();
-  const [loading, setLoading] = useState(false);
-  const [error, setError] = useState<SwapError>();
-  const [isTransactionPending, setPendingTransaction] = useState(false);
   const [lifeCycleStatus, setLifeCycleStatus] = useState<LifeCycleStatus>({
     statusName: 'init',
     statusData: {
       isMissingRequiredField: true,
       maxSlippage: config.maxSlippage,
     },
   }); // Component lifecycle
+
+  // Update lifecycle status, statusData will be persisted for the full lifeCycle
+  const updateLifeCycleStatus = useCallback(
+    (newStatus: LifeCycleStatusUpdate) => {
+      setLifeCycleStatus((prevStatus: LifeCycleStatus) => {
+        // do not persist errors
+        const persistedStatusData =
+          prevStatus.statusName === 'error'
+            ? (({ error, code, message, ...statusData }) => statusData)(
+                prevStatus.statusData,
+              )
+            : prevStatus.statusData;
+        return {
+          statusName: newStatus.statusName,
+          statusData: {
+            ...persistedStatusData,
+            ...newStatus.statusData,
+          },
+        } as LifeCycleStatus;
+      });
+    },
+    [],
+  );
```

Comment:
This is the main update, creating a setState callback function here that persists all statusData except errors, this data can be overwritten, but will persist until init is called again.

Reply:
very very clever, and can't wait for next PRs to have this abstract even more, so we can use it in other components.